File: fixtures/enterprise_fintech_platform/mobile_ios_swift/Models/Account.swift
import Foundation

public enum AccountCategory: String, Codable, CaseIterable {
    case checking = "CHECKING"
    case savings = "SAVINGS"
    case credit = "CREDIT"
    case wealth = "WEALTH"
}

public struct AccountItem: Identifiable, Codable {
    public let id: UUID
    public let accountNumber: String
    public let holderName: String
    public let balance: Double
    public let currency: String
    public let category: AccountCategory
    public let isLocked: Bool
    public let updatedAt: Date

    public init(
        id: UUID = UUID(),
        accountNumber: String,
        holderName: String,
        balance: Double,
        currency: String = "USD",
        category: AccountCategory,
        isLocked: Bool = false,
        updatedAt: Date = Date()
    ) {
        self.id = id
        self.accountNumber = accountNumber
        self.holderName = holderName
        self.balance = balance
        self.currency = currency
        self.category = category
        self.isLocked = isLocked
        self.updatedAt = updatedAt
    }
}

File: fixtures/enterprise_fintech_platform/mobile_android_kotlin/models/Account.kt
package com.navigator.fintech.mobile.models

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.math.BigDecimal
import java.util.UUID

enum class AccountType {
    CHECKING,
    SAVINGS,
    INVESTMENT,
    CREDIT
}

@Entity(tableName = "bank_accounts")
data class BankAccount(
    @PrimaryKey val id: String = UUID.randomUUID().toString(),
    val accountNumber: String,
    val accountHolderName: String,
    val accountType: AccountType,
    val balance: BigDecimal,
    val currency: String = "USD",
    val isFrozen: Boolean = false,
    val createdAtEpochMs: Long = System.currentTimeMillis()
)
